docs/content/docs/dev/table/sql/analyze.md

```diff
@@ -0,0 +1,363 @@
+tableEnv.executeSql(
+  "CREATE TABLE Store (" +
+          " `id` BIGINT NOT NULl," +
+          " location VARCHAR(32)," +
+          " owner VARCHAR(32)" +
+          ") with (...)");
+
+// register a partition table named "Orders"
+tableEnv.executeSql(
+  "CREATE TABLE Orders (" +
+          " `id` BIGINT NOT NULl," +
+          " product VARCHAR(32)," +
+          " amount INT," +
+          " `sold_year` BIGINT", +
+          " `sold_month` BIGINT", +
+          ") PARTITIONED BY (`sold_year`, `sold_month`) "
+") with (...)");
+
+// Non-partition table, collect row count.
+tableEnv.executeSql("ANALYZE TABLE Store COMPUTE STATISTICS");
+
+// Non-partition table, collect row count and statistics for all columns.
+tableEnv.executeSql("ANALYZE TABLE Store COMPUTE STATISTICS FOR ALL COLUMNS");
+
+// Non-partition table, collect row count and statistics for column `location`.
+tableEnv.executeSql("ANALYZE TABLE Store COMPUTE STATISTICS FOR COLUMNS(location)");
+
+
+// Suppose table "Orders" has 4 partitions with specs:
+// Partition1 : (sold_year='2022', sold_month='8')
+// Partition2 : (sold_year='2022', sold_month='9')
+// Partition3 : (sold_year='2021', sold_month='8')
+// Partition3 : (sold_year='2021', sold_month='9')
+
+
+// Partition table, collect row count for Partition1.
+tableEnv.executeSql("ANALYZE TABLE Orders PARTITION (sold_year='2022', sold_month='8') COMPUTE STATISTICS");
+
+// Partition table, collect row count for Partition1 and Partition2.
+tableEnv.executeSql("ANALYZE TABLE Orders PARTITION (sold_year='2022', sold_month) COMPUTE STATISTICS");
+
+// Partition table, collect row count for all partitions.
+tableEnv.executeSql("ANALYZE TABLE Orders COMPUTE STATISTICS");
+
+// Partition table, collect row count and statistics for all columns on partition1.
+tableEnv.executeSql("ANALYZE TABLE Orders PARTITION (sold_year='2022', sold_month='8') COMPUTE STATISTICS FOR ALL COLUMNS");
+
+// Partition table, collect row count and statistics for all columns on partition1 and partition2.
+tableEnv.executeSql("ANALYZE TABLE Orders PARTITION (sold_year='2022', sold_month) COMPUTE STATISTICS FOR ALL COLUMNS");
+
+// Partition table, collect row count and statistics for all columns on all partitions.
+tableEnv.executeSql("ANALYZE TABLE Orders COMPUTE STATISTICS FOR ALL COLUMNS");
+
+// Partition table, collect row count and statistics for column `amount` on partition1.
+tableEnv.executeSql("ANALYZE TABLE Orders PARTITION (sold_year='2022', sold_month='8') COMPUTE STATISTICS FOR COLUMNS(amount)");
+
+// Partition table, collect row count and statistics for `amount` and `product` on partition1 and partition2.
+tableEnv.executeSql("ANALYZE TABLE Orders PARTITION (sold_year='2022', sold_month) COMPUTE STATISTICS FOR COLUMNS(amount, product)");
+
+// Partition table, collect row count and statistics for column `amount` and `product` on all partitions.
+tableEnv.executeSql("ANALYZE TABLE Orders COMPUTE STATISTICS FOR COLUMNS(amount, product)");
```

Comment:
the partition keys should be specific for a partition table.

Reply:
> the partition keys should be specific for a partition table.

done!